internal/component/database_observability/mysql/collector/query_sample.go

```diff
@@ -300,3 +299,75 @@ func (c *QuerySample) fetchQuerySamples(ctx context.Context) error {
 
 	return nil
 }
+
+func (c *QuerySample) calculateTimestamp(serverStartTime, timer float64) uint64 {
```

Comment:
Wonder if it might help to distinguish the type of timestamp here. Perhaps we can name this to something like `calculateWallTime(serverStartTime, timestamp)`